src/Proto3/Suite/DotProto/Generate.hs

```diff
@@ -769,11 +773,13 @@ messageInstD ctxt parentIdent msgIdent messageParts =
         | QualifiedField _ fieldInfo <- qualifiedFields ]
 
      dotProtoE <- HsList <$> sequence
-         [ dpTypeE dpType >>= \typeE ->
-             pure (apply dotProtoFieldC [ fieldNumberE fieldNum, typeE
-                                        , dpIdentE fieldIdent
-                                        , HsList (map optionE options)
-                                        , maybeE (HsLit . HsString) comments ])
+         [ dpTypeE dpType <&> \typeE ->
```

Comment:
Generous use of `let` bindings can mitigate dependence on more exotic operators.  For example:

```haskell
-- I like to use `adapt` when I'm not sure what to name a function, sort of like how
-- people use `go` for loops
[ let adapt typeE = apply dotProtoFieldC ...
  in  fmap adapt (dpTypeE dpType)
, ...
```